tests/unit/testplan/common/utils/test_convert.py

```diff
@@ -0,0 +1,40 @@
+import pytest
+
+from testplan.common.utils.convert import flatten_dict_comparison
+
+
+comparisons = [
+    # expected = {38: 5, 555: [{600: 'A'}], 54: 2, 851: 4}
+    # actual = {38: 4, 54: 2, 20001: "SAMPLE"}
+    [
+        (38, "f", (0, "int", "5"), (0, "int", "4")),
+        (555, "f", (1, [(2, [(600, (0, "str", "A"))])]), (0, None, "ABSENT")),
+        (54, "p", (0, "int", "2"), (0, "int", "2")),
+        (851, "f", (0, "int", "4"), (0, None, "ABSENT")),
+        (20001, "f", (0, None, "ABSENT"), (0, "str", "SAMPLE")),
+    ],
+]
+result_tables = [
+    # expansion and extraction of left side from below comparison would
+    # produce an placeholder row
+    # ((555,), 1, 'ABSENT', 'f', '')
+    # which should not be popped together with a right side entry
+    # unless the same repeating group is present
+    [
+        [0, 38, "Failed", ("int", "5"), ("int", "4")],
+        [0, 555, "Failed", "", (None, "ABSENT")],
+        [0, "", "Failed", "", None],
+        [1, 600, "Failed", ("str", "A"), None],
+        [0, 54, "Passed", ("int", "2"), ("int", "2")],
+        [0, 851, "Failed", ("int", "4"), (None, "ABSENT")],
+        [0, 20001, "Failed", (None, "ABSENT"), ("str", "SAMPLE")],
+    ],
+]
+
+
+@pytest.mark.parametrize(
+    "comparison, result_table", zip(comparisons, result_tables)
+)
```

Comment:
Right now this is a single parameter pair, but this may be good to extend with more examples.